src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py

```diff
@@ -0,0 +1,258 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from binascii import hexlify
+from os import urandom
+
+from knack.util import CLIError
+from knack.log import get_logger
+
+from msrestazure.tools import is_valid_resource_id, parse_resource_id
+
+from azure.cli.core.commands import LongRunningOperation
+from azure.cli.core.azclierror import (ResourceNotFoundError, RequiredArgumentMissingError,
+                                       MutuallyExclusiveArgumentError)
+
+from azure.cli.command_modules.appservice._appservice_utils import _generic_site_operation
+from azure.cli.command_modules.appservice.utils import (_list_app, _rename_server_farm_props)
+from azure.cli.command_modules.appservice._client_factory import web_client_factory
+from azure.cli.command_modules.appservice.custom import (
+    _format_fx_version,
+    _get_extension_version_functionapp,
+    get_app_insights_key,
+    parse_docker_image_name,
+    list_consumption_locations,
+    is_plan_elastic_premium,
+    _validate_and_get_connection_string,
+    update_container_settings_functionapp,
+    try_create_application_insights,
+    _set_remote_or_local_git,
+    _fill_ftp_publishing_url)
+
+from ._constants import (FUNCTIONS_VERSION_TO_SUPPORTED_RUNTIME_VERSIONS,
+                         FUNCTIONS_VERSION_TO_DEFAULT_RUNTIME_VERSION,
+                         DOTNET_RUNTIME_VERSION_TO_DOTNET_LINUX_FX_VERSION)
+
+
+logger = get_logger(__name__)
+
+
+def create_logicapp(cmd, resource_group_name, name, storage_account, plan=None,
+                    os_type=None, consumption_plan_location=None,
+                    app_insights=None, app_insights_key=None, disable_app_insights=None,
+                    deployment_source_url=None, deployment_source_branch='master', deployment_local_git=None,
+                    docker_registry_server_password=None, docker_registry_server_user=None,
+                    deployment_container_image_name=None, tags=None):
+    # pylint: disable=too-many-statements, too-many-branches, too-many-locals
+    functions_version = '3'
+    runtime = None
+    runtime_version = None
+
+    if consumption_plan_location or not deployment_container_image_name:
+        runtime = 'node'
+        runtime_version = '12'
```

Comment:
>
functions_version = '3'
runtime_version = '12'

Do we need to consider putting these constants in file `_constants.py `?